frontend/src/pages/users/team/index.tsx

```diff
@@ -1,33 +1,47 @@
-import { DataTable } from "@/components/ui/data-table";
+import { DataTable } from "@/components/paul/data-table";
 import { useValidatedProps } from "@/hooks/use-validated-props.ts";
-import BaseLayout from "@/layouts/base-layout.tsx";
+import BaseLayout from "@/layouts/base-layout";
 import { useMemo } from "react";
 import { Columns } from "./columns";
 import { useTranslation } from "react-i18next";
 import { ArrowPathIcon, UserPlusIcon } from "@heroicons/react/20/solid";
-import { InputSearch } from "@/components/ui/input-search.tsx";
+import { InputSearch } from "@/components/ui/input-search";
 import { FunnelIcon } from "@heroicons/react/24/outline";
-import { Button } from "@/components/ui/button.tsx";
+import { Button } from "@/components/ui/button";
 import userEmptyImage from "@/assets/user-empty.svg";
-import PaginationElided from "@/components/ui/pagination.tsx";
-import {
-  type TeamPageProps,
-  TeamPagePropsStruct
-} from "@/pages/users/team/team-page-props-struct.ts";
-import type { Pagination } from "@/types/pagination.ts";
+import { type TeamPageProps, TeamPagePropsStruct } from "@/pages/users/team/team-page-props-struct.ts";
+import { PaginationFooter } from "@/components/paul/pagination-footer";
+import { router } from "@inertiajs/react";
 
 
 export default function TeamPage() {
   const {
-    props: {users, user_count, pagination},
+    props: {users, user_count, pagination, search_query},
   } = useValidatedProps<TeamPageProps>(TeamPagePropsStruct);
 
   const {t} = useTranslation();
 
-  // Pass t to Columns to avoid calling hooks inside useMemo
+  const [pageSize, setPageSize] = useMemo(() => {
```

Comment:
setting the correct page size query param is not the responsibility of this component. Remove this from here and implement the url search params the same way as in `handlePageChange` from `ui/pagination.tsx`